### PR TITLE
Planner: Add variable to control automated TopN count

### LINKED ISSUE #53111: TopN statistics are collected unnecessarily for non-skewed values on small tables
## Enhancement

Current ANALYZE will collect by default the top 500 values for an interesting column (indexed column or predicate column). For tables that are sampled, then there is pruning logic to remove statistics that are not skewed. For tables that are NOT sampled (smaller tables) - the pruning logic is not invoked.

Below shows an example where values are collected with a count of 1 - which are not skewed.

```
tidb> show stats_topn;
+---------+------------+----------------+-------------+----------+------------+-------+
| Db_name | Table_name | Partition_name | Column_name | Is_index | Value      | Count |
+---------+------------+----------------+-------------+----------+------------+-------+
| test    | t2         |                | a           |        0 | 73         |    16 |
| test    | t2         |                | a           |        0 | 74         |    16 |
| test    | t2         |                | a           |        0 | 75         |    16 |
| test    | t2         |                | a           |        0 | 76         |    16 |
......
| test    | t2         |                | a           |        0 | 101        |     1 |
| test    | t2         |                | a           |        0 | 102        |     1 |
| test    | t2         |                | a           |        0 | 103        |     1 |
| test    | t2         |                | a           |        0 | 104        |     1 |
......
```

### PR DESCRIPTION
**Pull request by terry1purcell**
### What problem does this PR solve?

Issue Number: close #53111 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1054,6 +1054,20 @@ var defaultSysVars = []*SysVar{
 			return normalizedValue, nil
 		},
 	},
+	{
```
Comment (hawkingrei) [inline pkg/sessionctx/variable/sysvar.go:1057]: I recommend incorporating the addition of the switch together with the main functionality. Adding a switch doesn't require many lines and won't affect the review process.

**Comment by winoros** [comment]
Is this pull related to the previous one https://github.com/pingcap/tidb/pull/53035?
Do we find out some regression about the previous one so that we add this variable?

**Comment by terry1purcell** [comment]
There has not been a request by a customer to deliver a variable. So I am closing this PR, and will reopen if we receive a customer request.

**Pull request closed without merging**